src/dbnode/encoding/scheme.go

```diff
@@ -153,7 +168,22 @@ func (tes *timeEncodingScheme) Buckets() []TimeBucket     { return tes.buckets }
 func (tes *timeEncodingScheme) DefaultBucket() TimeBucket { return tes.defaultBucket }
 
 // TimeEncodingSchemes defines the time encoding schemes for different time units.
-type TimeEncodingSchemes map[xtime.Unit]TimeEncodingScheme
+type TimeEncodingSchemes []TimeEncodingScheme
+
+// SchemeForUnit returns the corresponding TimeEncodingScheme for the provided unit.
+// Returns false if the unit does not match a scheme or is invalid.
+func (s TimeEncodingSchemes) SchemeForUnit(u xtime.Unit) (TimeEncodingScheme, bool) {
+	if !u.IsValid() {
+		return nil, false
+	}
+
+	scheme := s[u]
```

Comment:
Do you think it's worth checking that `0 <= u < len(s)` or is that being a bit too defensive?

Reply:
Hmm yeah - I'm also not totally sure. We always new the schemes array via `newTimeEncodingSchemes` which ensures the array is size of `unitCount`, which means we know `u` will be valid in it. But since technically `TimeEncodingSchemes` could be created manually maybe it is safest to just check to avoid an invalid index err.